build.rs

```diff
@@ -31,65 +58,104 @@ struct Hsl {
+struct AnsiColor {
+    name: String,
+    rgb: Rgb,
+    hsl: Hsl,
+    code: u8,
 }
 
-type Palette = HashMap<String, Flavor>;
-
 fn main() -> Result<(), Box<dyn Error>> {
     let out_dir = PathBuf::from(&env::var("OUT_DIR")?);
     let codegen_path = out_dir.join("generated_palette.rs");
     let mut code_writer = BufWriter::new(File::create(codegen_path)?);
 
     let palette: Palette =
         serde_json::from_reader(BufReader::new(File::open("src/palette.json")?))?;
-    let sample_flavor = palette.values().next().expect("at least one flavor");
-
-    let tokens = [
-        make_flavorcolors_struct(sample_flavor),
-        make_flavorcolors_all_impl(sample_flavor),
-        make_colorname_enum(sample_flavor),
-        make_colorname_index_impl(sample_flavor),
-        make_colorname_display_impl(sample_flavor),
-        make_colorname_identifier_impl(sample_flavor),
-        make_colorname_fromstr_impl_tokens(sample_flavor),
-        make_palette_const(&palette),
+    let sample_flavor = palette
+        .flavors
+        .values()
+        .next()
+        .expect("at least one flavor");
+
+    let flavor_tokens = [
+        // Colors
+        make_flavor_colors_struct(sample_flavor),
+        make_flavor_colors_all_impl(sample_flavor),
+        // ANSI Colors
+        make_flavor_ansi_colors_struct(sample_flavor),
+        make_flavor_ansi_colors_all_impl(sample_flavor),
+        // ANSI Color Pairs
+        make_flavor_ansi_color_pairs_struct(sample_flavor),
+        make_flavor_ansi_color_pairs_all_impl(sample_flavor),
+    ];
+    let color_tokens = [
+        make_color_name_enum(sample_flavor),
+        make_color_name_index_impl(sample_flavor),
+        make_color_name_display_impl(sample_flavor),
+        make_color_name_identifier_impl(sample_flavor),
+        make_color_name_fromstr_impl_tokens(sample_flavor),
+    ];
+    let ansi_color_tokens = [
+        make_ansi_color_name_enum(sample_flavor),
+        make_ansi_color_name_index_impl(sample_flavor),
+        make_ansi_color_name_display_impl(sample_flavor),
+        make_ansi_color_name_identifier_impl(sample_flavor),
+        make_ansi_color_name_fromstr_impl_tokens(sample_flavor),
     ];
-    let ast = syn::parse2(tokens.into_iter().collect())?;
+    let ansi_color_pair_tokens = [
+        make_ansi_color_pair_name_enum(sample_flavor),
+        make_ansi_color_pair_name_index_impl(sample_flavor),
+        make_ansi_color_pair_name_display_impl(sample_flavor),
+        make_ansi_color_pair_name_identifier_impl(sample_flavor),
+        make_ansi_color_pair_name_fromstr_impl_tokens(sample_flavor),
+    ];
+    let palette_tokens = [make_palette_const(&palette)];
+
+    let ast = syn::parse2(
+        [
+            &flavor_tokens[..],
+            &color_tokens[..],
+            &ansi_color_tokens[..],
+            &ansi_color_pair_tokens[..],
+            &palette_tokens[..],
+        ]
+        .concat()
+        .into_iter()
+        .collect(),
+    )?;
```

Comment:
I split these out into separate arrays while implementing the methods to keep myself straight but I think it's good to merge them back into one array.